src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -206,6 +207,24 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     [accountSpecifiersList, dispatch],
   )
 
+  /**
+   * refetch an account given a newly confirmed txid
+   */
+  const refetchStakingDataByTxId = useCallback(
+    (txId: TxId) => {
+      // the accountSpecifier the tx came from
+      const { txAccountSpecifier } = deserializeUniqueTxId(txId)
```

Comment:
`txId` should always be defined/a string with length and thus we don't technically need it, but I'd add it just for that bit of extra safety and to lead by example in case someone's looking in this implementation and want to dispatch `stakingDataApi.endpoints.getStakingData.initiate` somewhere else in the codebase, where `accountSpecifier` might be an empty string.

```suggestion
      const { txAccountSpecifier } = deserializeUniqueTxId(txId)
      if (txAccountSpecifier.length) return
```